app/controllers/application_controller.rb

```diff
@@ -1,5 +1,12 @@
 class ApplicationController < ActionController::Base
-  def current_user
-    User.first
+  protect_from_forgery with: :exception
+
+  before_action :authenticate_user!, :update_allowed_parameters, if: :devise_controller?
+
```

Comment:
- [ ] **[OPTIONAL]** Users can still view the users' list page when they sign up and have not activated their emails. It would be nice to prevent users from visiting any page if they are not fully signed up yet. I think the `if:` condition on the same line as `authenticate_user!` is causing this.

  ```
  before_action :authenticate_user!
  before_action :update_allowed_parameters, if: :devise_controller?
  ```